owasp-top10-2017-apps/a5/tictactoe/README.md

```diff
@@ -0,0 +1,271 @@
+
+# Tic-Tac-Toe
+
+  
+
+<p  align="center">
+
+<img  src="images/a5-banner.png"/>
+
+</p>
+
+  
+
+Tic-Tac-Toe is a simple Nodejs web application that contains an example of a Broken Access Control vulnerability and its main goal is to describe how a malicious user could exploit it.
+
+  
+
+## Index
+
+  
+
+-  [Definition](#what-is-broken-access-control)
+
+-  [Setup](#setup)
+
+-  [Attack narrative](#attack-narrative)
+
+-  [Objectives](#secure-this-app)
+
+-  [Solutions](#pr-solutions)
+
+-  [Contributing](#contributing)
+
+  
+
+## What is Broken Access Control?
+
+  
+
+Restrictions on what authenticated users are allowed to do are often not properly enforced. Attackers can exploit these flaws to access unauthorized functionality and/or data, such as access to other users' accounts, view sensitive files, modify other users’ data, change access rights, etc.
+
+  
+
+The main goal of this app is to discuss how **Broken Access Control** vulnerabilities can be exploited and to encourage developers to send secDevLabs Pull Requests on how they would mitigate these flaws.
+
+  
+
+## Setup
+
+  
+
+To start this intentionally **insecure application**, you will need [Docker][Docker Install] and [Docker Compose][Docker Compose Install]. After forking [secDevLabs](https://github.com/globocom/secDevLabs), you must type the following commands to start:
+
+  
+
+```sh
+
+cd secDevLabs/owasp-top10-2017-apps/a5/tictactoe
+
+```
+
+  
+
+```sh
+
+make install
+
+```
+
+  
+
+Then simply visit [localhost:10005][App] ! 😆
+
+  
+
+## Get to know the app 💵
+
+  
+
+To properly understand how this application works, you can follow this step:
+
+  
+
+- Try registering a user, sign in, play the game and see your statistics!
+
+  
+
+## Attack narrative
+
+  
+
+Now that you know the purpose of this app, what could go wrong? The following section describes how an attacker could identify and eventually find sensitive information about the app or its users. We encourage you to follow these steps and try to reproduce them on your own to better understand the attack vector! 😜
+
+  
+
+### 👀
+
+  
+
+In order to better understand how this app works, two users, `fry123` and `bender`, were created using web interface by visiting localhost:10005/create as shown below:
```

Comment:
Hmm, what do you think of this? 

`To better understand how this attack works, two users, `fry123` and `bender`, were created using the web interface by visiting localhost:10005/create as shown below:`

Reply:
Yes, it makes sense. Changed!